atest/testdata/test_libraries/LibraryDecorator.py

```diff
@@ -0,0 +1,15 @@
+from robot.api.deco import keyword, library
+
+
+@library(scope='TEST SUITE', version='1.2.3')
+class LibraryDecorator(object):
+
+    def __init__(self):
+        self.invalid = 'This method is not a keyword.'
+
+    def library_decorator_disables_public_methods(self):
+        print(self.invalid)
+
+    @keyword(name="Decorated Method Is Keyword")
+    def decorated_method(self):
+        print(self.invalid)
```

Comment:
It's a bit strange to print `self.invalid` here when this is valid usage. I doubt `self.invalid` is needed in general. Could just print something relevant here and in the other use something like `raise RuntimeError("Should not be executed!")`.